// fighting-design/es/submenu/src/index.js
import { defineComponent as h, inject as k, ref as C, resolveComponent as $, openBlock as i, createElementBlock as m, normalizeClass as c, unref as e, createBlock as g, withCtx as u, renderSlot as l, createElementVNode as s, createCommentVNode as w, createVNode as a } from "vue";
import { Props as E } from "./index3.js";
import { MENU_PROVIDE_KEY as I } from "../../menu/src/index3.js";
import { FDropdown as F } from "../../dropdown/index.js";
import { FSvgIcon as _ } from "../../svg-icon/index.js";
import d from "../../_svg/f-icon-chevron-down/index.js";
import { useRun as N } from "../../_hooks/use-run/index.js";
const S = { class: "f-submenu__title" }, V = { class: "f-submenu__title-text" }, y = {
  key: 1,
  role: "none",
  class: "f-submenu__inline"
}, B = {
  role: "none",
  class: "f-submenu__title-text"
}, D = {
  role: "none",
  class: "f-submenu__content"
};
var Y = /* @__PURE__ */ h({
  name: "FSubmenu",
  __name: "submenu",
  props: E,
  setup(f) {
    const r = f, { run: p } = N(), n = k(I, null), t = C(r.opened), v = (o) => {
      r.disabled || (t.value = !t.value, n && p(n.onSubmenuClick, t.value, o));
    };
    return (o, j) => {
      const b = $("f-collapse-animation");
      return i(), m("div", {
        role: "menuitem",
        class: c([
          "f-submenu",
          {
            [`f-submenu__${e(n) && e(n).mode.value}`]: e(n) && e(n).mode.value,
            "f-submenu__disabled": o.disabled
          }
        ])
      }, [
        e(n) && e(n).mode.value !== "inline" ? (i(), g(e(F), {
          key: 0,
          trigger: "click",
          spacing: 5
        }, {
          content: u(() => [
            l(o.$slots, "default")
          ]),
          default: u(() => [
            s("div", S, [
              s("div", V, [
                o.$slots.title ? l(o.$slots, "title", { key: 0 }) : w("", !0)
              ]),
              a(e(_), { icon: e(d) }, null, 8, ["icon"])
            ])
          ]),
          _: 3
        })) : (i(), m("div", y, [
          s("div", {
            role: "none",
            class: "f-submenu__title",
            onClick: v
          }, [
            s("div", B, [
              l(o.$slots, "title")
            ]),
            a(e(_), {
              class: c([
                "f-submenu__arrow-icon",
                { "f-submenu__arrow-icon-animation": t.value }
              ]),
              icon: e(d)
            }, null, 8, ["class", "icon"])
          ]),
          s("div", D, [
            a(b, {
              opened: t.value,
              "animation-time": 0.27
            }, {
              default: u(() => [
                l(o.$slots, "default")
              ]),
              _: 3
            }, 8, ["opened"])
          ])
        ]))
      ], 2);
    };
  }
});
export {
  Y as default
};
