// fighting-design/es/textarea/src/index3.js
import { setStringNumberProp as r, setBooleanProp as e, setNumberProp as a, setStringProp as t, setFunctionProp as o } from "../../_utils/props/index.js";
const i = {
  /** 绑定值 */
  modelValue: {
    ...r(),
    required: !0
  },
  /**
   * 元素的输入文本的行数
   *
   * @see rows https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/textarea#attr-rows
   */
  rows: r(),
  /**
   * 是否禁用
   *
   * @see disabled https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/textarea#attr-disabled
   */
  disabled: e(),
  /** 是否自适应高度 */
  autoHeight: e(),
  /**
   * 最大输入长度
   *
   * @see maxLength https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/textarea#attr-maxlength
   */
  maxLength: a(),
  /**
   * 是否自动获取焦点
   *
   * @see autofocus https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/textarea#attr-autofocus
   */
  autofocus: e(),
  /**
   * 原生 name 属性
   *
   * @see name https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/textarea#attr-name
   */
  name: t("f-textarea"),
  /**
   * 占位符
   *
   * @see placeholder https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/textarea#attr-placeholder
   */
  placeholder: t(),
  /**
   * 是否只读
   *
   * @see readonly https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/textarea#attr-readonly
   */
  readonly: e(),
  /**
   * 设置字体大小
   *
   * @see font-size https://developer.mozilla.org/zh-CN/docs/Web/CSS/font-size
   */
  fontSize: r(),
  /** input 事件触发的回调 */
  onInput: o(),
  /**
   * 拉伸方式
   *
   * @see resize https://developer.mozilla.org/zh-CN/docs/Web/CSS/resize
   * @values none both horizontal vertical
   */
  resize: t(void 0, (n) => ["none", "both", "horizontal", "vertical"].includes(n)),
  /** 是否可清空 */
  clear: e(),
  /** 绑定值发生变化时触发的回调 */
  onChange: o(),
  /** 失去焦点时触发的回调 */
  onBlur: o(),
  /** 获取焦点时触发的回调 */
  onFocus: o(),
  /** 按下回车触发的 */
  onEnter: o()
};
export {
  i as Props
};
