// fighting-design/es/sticky-card/src/index.js
import { defineComponent as k, ref as v, computed as C, unref as s, openBlock as a, createElementBlock as l, normalizeStyle as x, renderSlot as i, createCommentVNode as h, createElementVNode as n, createVNode as T, withCtx as g, normalizeClass as S, toDisplayString as b } from "vue";
import { Props as N } from "./index3.js";
import { FCollapseAnimation as V } from "../../collapse-animation/index.js";
import { useGlobal as $ } from "../../_hooks/use-global/index.js";
import { useList as z } from "../../_hooks/use-list/index.js";
import { useRun as B } from "../../_hooks/use-run/index.js";
const E = {
  key: 0,
  class: "f-sticky-card__source"
}, F = { class: "f-sticky-card__box" }, L = { class: "f-sticky-card__option-text" };
var R = /* @__PURE__ */ k({
  name: "FStickyCard",
  __name: "sticky-card",
  props: N,
  setup(d) {
    const o = d, { getLang: p } = $(), { run: u } = B(), { styles: m } = z(o, "sticky-card"), e = v(o.open), _ = () => {
      e.value = !e.value, u(e.value ? o.onClose : o.onOpen, e.value);
    }, f = C(() => {
      const { openText: t, closeText: r } = o, c = p("stickyCard").value;
      return s(e) ? t || c.openText : r || c.closeText;
    }), y = m(["borderColor"]);
    return (t, r) => (a(), l("div", {
      class: "f-sticky-card",
      style: x(s(y))
    }, [
      t.$slots.source ? (a(), l("div", E, [
        i(t.$slots, "source")
      ])) : h("", !0),
      n("div", F, [
        T(s(V), { opened: e.value }, {
          default: g(() => [
            i(t.$slots, "default")
          ]),
          _: 3
        }, 8, ["opened"])
      ]),
      n("div", {
        class: S(["f-sticky-card__option", { "f-sticky-card__option-open": e.value }]),
        onClick: _
      }, [
        n("span", L, b(f.value), 1)
      ], 2)
    ], 4));
  }
});
export {
  R as default
};
