// fighting-design/es/radio-group/src/index.js
import { defineComponent as d, mergeModels as p, useModel as c, provide as f, reactive as g, toRefs as v, openBlock as y, createElementBlock as R, normalizeClass as _, unref as t, normalizeStyle as k, renderSlot as B } from "vue";
import { Props as G, RADIO_GROUP_PROPS_kEY as P } from "./index3.js";
import { useList as S } from "../../_hooks/use-list/index.js";
import { useRun as V } from "../../_hooks/use-run/index.js";
var O = /* @__PURE__ */ d({
  name: "FRadioGroup",
  __name: "radio-group",
  props: /* @__PURE__ */ p(G, {
    modelValue: { type: Boolean, default: "", type: [String, Number, Boolean] },
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(r) {
    const o = r, s = c(r, "modelValue"), { run: a } = V(), { styles: l, classes: n } = S(o, "radio-group"), u = (e) => {
      s.value = e, a(o.onChange, e);
    }, i = n(["vertical", "background", "size"], "f-radio-group"), m = l(["columnGap", "rowGap"]);
    return f(
      P,
      g({
        ...v(o),
        changeEvent: u
      })
    ), (e, h) => (y(), R("div", {
      role: "radiogroup",
      class: _(t(i)),
      style: k(t(m))
    }, [
      B(e.$slots, "default")
    ], 6));
  }
});
export {
  O as default
};
