// fighting-design/es/dropdown/src/index.js
import { defineComponent as r, openBlock as a, createElementBlock as t, createVNode as i, unref as d, withCtx as o, renderSlot as n } from "vue";
import { Props as s } from "./index3.js";
import { FTrigger as p } from "../../trigger/index.js";
const l = { class: "f-dropdown" };
var h = /* @__PURE__ */ r({
  name: "FDropdown",
  __name: "dropdown",
  props: s,
  setup(g) {
    return (e, u) => (a(), t("div", l, [
      i(d(p), {
        trigger: e.trigger,
        disabled: e.disabled,
        spacing: e.spacing,
        arrow: e.arrow,
        width: e.width,
        "enter-duration": e.enterDuration,
        "leave-duration": e.leaveDuration,
        "on-change": e.onChange,
        "on-open": e.onOpen,
        "on-close": e.onClose
      }, {
        content: o(() => [
          n(e.$slots, "content")
        ]),
        default: o(() => [
          n(e.$slots, "default")
        ]),
        _: 3
      }, 8, ["trigger", "disabled", "spacing", "arrow", "width", "enter-duration", "leave-duration", "on-change", "on-open", "on-close"])
    ]));
  }
});
export {
  h as default
};
