// fighting-design/es/avatar/src/index3.js
import { FIGHTING_FIT as n, FIGHTING_SIZE as s } from "../../_tokens/attrs/index.js";
import { setStringProp as r, setObjectProp as u, setBooleanProp as t, setStringNumberProp as e, setFunctionProp as i } from "../../_utils/props/index.js";
import { isString as p, isNumber as d } from "../../_utils/is/index.js";
const l = {
  /** 图片路径 */
  src: r(),
  /** 加载失败时要显示的地址 */
  errSrc: r(),
  /** 图标头像 */
  icon: u(),
  /**
   *原生 alt 属性
   *
   * @see alt https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/img#attr-alt
   */
  alt: r(),
  /** 是否为圆角 */
  round: t(),
  /** 是否懒加载头像 */
  lazy: t(),
  /**
   * 如何适应容器
   *
   * 原生样式属性
   *
   * @see object-fit https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit#try_it
   * @values fill contain cover none scale-down
   */
  fit: r(void 0, (o) => n.includes(o)),
  /**
   * 头像大小
   *
   * 可传入字符串使用内置大小
   *
   * 也可以传入数字，数字将自动转换为 px 单位
   *
   * @values large middle small mini
   * @default middle
   */
  size: {
    type: [String, Number],
    default: () => "middle",
    validator: (o) => p(o) ? s.includes(o) : d(o) ? o >= 1 : !1
  },
  /** 自定义背景色 */
  background: r(),
  /** 字体大小 */
  fontSize: e(),
  /** 字体的颜色 */
  fontColor: r(),
  /** 文字头像 */
  text: r(),
  /** 触发懒加载的距离 */
  rootMargin: e("100px"),
  /** 图片加载成功触发的回调 */
  onLoad: i(),
  /** 图片加载失败触发的回调 */
  onError: i()
};
export {
  l as Props
};
