// fighting-design/es/rate/src/index.js
import { defineComponent as C, mergeModels as h, useModel as y, ref as _, watch as k, computed as u, unref as l, openBlock as a, createElementBlock as n, createElementVNode as z, Fragment as M, renderList as g, normalizeClass as N, createVNode as F, createBlock as S, withCtx as B, createTextVNode as w, toDisplayString as b, createCommentVNode as E } from "vue";
import { Props as I } from "./index3.js";
import { FSvgIcon as L } from "../../svg-icon/index.js";
import { FText as R } from "../../text/index.js";
import T from "../../_svg/f-icon-star-b/index.js";
import { isNumber as $ } from "../../_utils/is/index.js";
import { useRun as A } from "../../_hooks/use-run/index.js";
const D = {
  class: "f-rate",
  role: "slider"
}, P = { class: "f-rate__list" }, j = ["onMouseover", "onClick"];
var W = /* @__PURE__ */ C({
  name: "FRate",
  __name: "rate",
  props: /* @__PURE__ */ h(I, {
    modelValue: { default: 0, type: Number },
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(s) {
    const o = s, m = y(s, "modelValue"), { run: i } = A(), r = _(o.modelValue), c = (e) => {
      o.readonly || (r.value = e);
    }, d = () => {
      o.readonly || (r.value = o.modelValue);
    }, f = (e) => {
      o.readonly || (r.value = e, m.value = e, i(o.onChange, e));
    };
    k(
      () => o.modelValue,
      () => {
        r.value = o.modelValue;
      }
    );
    const v = u(() => o.textArr[l(r) - 1]), p = u(() => $(o.max) ? o.max : 5);
    return (e, q) => (a(), n("div", D, [
      z("div", P, [
        (a(!0), n(M, null, g(p.value, (x, t) => (a(), n("div", {
          key: t,
          class: N(["f-rate__star", { "f-rate__star-readonly": e.readonly }]),
          onMouseout: d,
          onMouseover: (V) => c(t + 1),
          onClick: (V) => f(t + 1)
        }, [
          F(l(L), {
            size: e.size,
            icon: e.icon || l(T),
            color: r.value > t ? e.effectColor : e.invalidColor
          }, null, 8, ["size", "icon", "color"])
        ], 42, j))), 128))
      ]),
      e.textShow ? (a(), S(l(R), {
        key: 0,
        size: e.textSize,
        color: e.textColor
      }, {
        default: B(() => [
          w(b(v.value), 1)
        ]),
        _: 1
      }, 8, ["size", "color"])) : E("", !0)
    ]));
  }
});
export {
  W as default
};
