// fighting-design/es/button/src/index.js
import { defineComponent as L, ref as T, computed as $, openBlock as t, createElementBlock as d, normalizeClass as l, unref as o, normalizeStyle as m, createBlock as a, createCommentVNode as r, renderSlot as b, toRefs as g, reactive as A } from "vue";
import { Props as D } from "./index3.js";
import { FSvgIcon as s } from "../../svg-icon/index.js";
import { useButton as E } from "../../_hooks/use-button/index.js";
import y from "../../_svg/f-icon-loading-a/index.js";
import { useRipples as G } from "../../_hooks/use-ripples/index.js";
import { useGlobal as P } from "../../_hooks/use-global/index.js";
import { useRun as V } from "../../_hooks/use-run/index.js";
const j = ["href", "target"], q = ["disabled", "autofocus", "name", "type"];
var W = /* @__PURE__ */ L({
  name: "FButton",
  __name: "button",
  props: D,
  setup(k) {
    const n = k, { getType: I } = P(n), { run: v } = V(), { classList: u, style: f } = E(n), i = T(), c = (e) => {
      const { disabled: p, loading: h, ripples: B } = g(n);
      if (p.value || h.value) {
        e.preventDefault();
        return;
      }
      if (B.value) {
        const { ripplesColor: R, simple: z, text: F } = g(n), N = A({
          duration: 700,
          component: "f-button",
          className: "f-button__ripples",
          ripplesColor: R.value,
          simple: z.value,
          text: F.value,
          type: I()
        }), { runRipples: S } = G(
          e,
          i.value,
          N
        );
        S();
      }
      v(n.onClick, e);
    }, C = $(() => n.loading ? n.loadingIcon || y : n.beforeIcon);
    return (e, p) => e.href ? (t(), d("a", {
      key: 0,
      ref_key: "FButtonRef",
      ref: i,
      role: "link",
      tabindex: "0",
      class: l(o(u)),
      href: e.href,
      target: e.target,
      style: m(o(f)),
      onClick: c
    }, [
      e.loading || e.beforeIcon ? (t(), a(o(s), {
        key: 0,
        class: l({ "f-button__loading-animation": e.loading }),
        icon: C.value,
        size: 16
      }, null, 8, ["class", "icon"])) : r("", !0),
      b(e.$slots, "default"),
      e.afterIcon ? (t(), a(o(s), {
        key: 1,
        icon: e.afterIcon,
        size: 16
      }, null, 8, ["icon"])) : r("", !0)
    ], 14, j)) : (t(), d("button", {
      key: 1,
      ref_key: "FButtonRef",
      ref: i,
      role: "button",
      tabindex: "0",
      class: l(o(u)),
      disabled: e.disabled || e.loading,
      autofocus: e.autofocus,
      name: e.name,
      type: e.nativeType,
      style: m(o(f)),
      onClick: c
    }, [
      e.loading || e.beforeIcon ? (t(), a(o(s), {
        key: 0,
        class: l(["f-button_before-icon", { "f-button__loading-animation": e.loading }]),
        icon: e.loading ? e.loadingIcon || o(y) : e.beforeIcon,
        size: 16
      }, null, 8, ["class", "icon"])) : r("", !0),
      b(e.$slots, "default"),
      e.afterIcon ? (t(), a(o(s), {
        key: 1,
        class: "f-button_after-icon",
        icon: e.afterIcon,
        size: 16
      }, null, 8, ["icon"])) : r("", !0)
    ], 14, q));
  }
});
export {
  W as default
};
