// fighting-design/es/toolbar/src/index.js
import { defineComponent as p, useSlots as d, openBlock as _, createElementBlock as b, normalizeClass as C, unref as r, normalizeStyle as k, renderSlot as g } from "vue";
import { Props as h } from "./index3.js";
import { TOOLBAR_ITEM_CLASS_NAME as A, TOOLBAR_INDEX as L } from "../../_tokens/key/index3.js";
import { useList as S } from "../../_hooks/use-list/index.js";
import { useRun as x } from "../../_hooks/use-run/index.js";
var z = /* @__PURE__ */ p({
  name: "FToolbar",
  __name: "toolbar",
  props: h,
  setup(s) {
    const e = s, l = d(), { run: n } = x(), { classes: a, styles: c } = S(e, "toolbar"), i = (o) => {
      if (!l.default)
        return;
      const t = o.target;
      if (t.className === A) {
        const f = t.getAttribute(L);
        n(e.onClick, f, o);
      }
    }, m = a(["size", "round", "fixed"], "f-toolbar"), u = c(["textColor", "background", "width", "height"]);
    return (o, t) => (_(), b("div", {
      role: "toolbar",
      class: C(r(m)),
      style: k(r(u)),
      onClick: i
    }, [
      g(o.$slots, "default")
    ], 6));
  }
});
export {
  z as default
};
