// fighting-design/es/table/src/index.js
import { defineComponent as A, computed as F, ref as T, watch as P, openBlock as e, createElementBlock as t, normalizeClass as R, unref as i, normalizeStyle as j, createElementVNode as d, Fragment as u, createCommentVNode as a, renderList as c, createVNode as m, createBlock as p, resolveDynamicComponent as b, createTextVNode as v, toDisplayString as k, renderSlot as V, h as S } from "vue";
import { Props as q } from "./index3.js";
import { FEmpty as G } from "../../empty/index.js";
import { FCheckbox as f } from "../../checkbox/index.js";
import { isFunction as C, isArray as I } from "../../_utils/is/index.js";
import { useRun as J } from "../../_hooks/use-run/index.js";
import { useList as K } from "../../_hooks/use-list/index.js";
const M = { class: "f-table__container" }, O = {
  key: 0,
  class: "f-table__header"
}, Q = { class: "f-table__table" }, W = {
  key: 0,
  width: 50,
  span: 1
}, X = {
  key: 1,
  width: 50,
  span: 1
}, Y = ["width"], Z = ["align"], x = { key: 0 }, ee = { key: 1 }, te = { class: "f-table__body" }, le = {
  key: 0,
  class: "f-table__table"
}, oe = {
  key: 0,
  width: 50,
  span: 1
}, se = {
  key: 1,
  width: 50,
  span: 1
}, ae = ["width"], ne = ["align"], re = { key: 0 }, ue = { key: 1 }, de = ["align"], ie = { key: 0 }, he = { key: 1 }, ce = { key: 1 }, ye = { class: "f-table__table" }, _e = {
  key: 0,
  width: 50,
  span: 1
}, me = { key: 1 }, ke = ["width"], pe = ["align"], be = { key: 0 }, ve = { key: 1 }, fe = { class: "table__no-data" }, ge = { key: 1 };
var ze = /* @__PURE__ */ A({
  name: "FTable",
  __name: "table",
  props: q,
  setup(L) {
    const h = L, { run: U } = J(), { styles: z, classes: B } = K(h, "table"), N = (l, n, o, s) => l(S, n, o, s), g = (l, n, o) => l(S, n, o), $ = z(["zebraColor", "bgColor", "headBgColor", "height", "width"]), E = B(["border", "zebra"], "f-table"), w = F(() => !!(h.height && h.showHead)), r = T([]), _ = F({
      get: () => r.value.length ? (r.value || []).every((l) => l._select) : !1,
      set: (l) => {
        const n = (r.value || []).map((o) => (o._select = l, o));
        return r.value = n, D(), l;
      }
    });
    P(() => h.data, () => {
      if (I(h.data) || (r.value = []), h.select) {
        const l = h.data.map((n) => ({
          ...n,
          _select: !1
        }));
        r.value = l;
        return;
      }
      r.value = h.data;
    }, { immediate: !0 });
    const D = () => {
      const l = (r.value || []).filter((n) => n._select);
      U(h.onSelect, l);
    };
    return (l, n) => (e(), t("div", {
      role: "table",
      class: R(i(E)),
      style: j(i($))
    }, [
      d("div", M, [
        l.columns || r.value ? (e(), t(u, { key: 0 }, [
          w.value ? (e(), t("div", O, [
            d("table", Q, [
              d("colgroup", null, [
                l.select ? (e(), t("col", W)) : a("", !0),
                l.num ? (e(), t("col", X)) : a("", !0),
                (e(!0), t(u, null, c(l.columns, (o, s) => (e(), t("col", {
                  key: s,
                  width: o.width
                }, null, 8, Y))), 128))
              ]),
              d("thead", { align: l.align }, [
                d("tr", null, [
                  l.select ? (e(), t("th", x, [
                    m(i(f), {
                      modelValue: _.value,
                      "onUpdate:modelValue": n[0] || (n[0] = (o) => _.value = o),
                      disabled: !r.value.length
                    }, null, 8, ["modelValue", "disabled"])
                  ])) : a("", !0),
                  l.num ? (e(), t("th", ee, "#")) : a("", !0),
                  (e(!0), t(u, null, c(l.columns, (o, s) => (e(), t("th", { key: s }, [
                    i(C)(o.title) ? (e(), p(b(g(o.title, o, s)), { key: 0 })) : (e(), t(u, { key: 1 }, [
                      v(k(o.title), 1)
                    ], 64))
                  ]))), 128))
                ])
              ], 8, Z)
            ])
          ])) : a("", !0),
          d("div", te, [
            r.value && r.value.length ? (e(), t("table", le, [
              d("colgroup", null, [
                l.select ? (e(), t("col", oe)) : a("", !0),
                l.num ? (e(), t("col", se)) : a("", !0),
                (e(!0), t(u, null, c(l.columns, (o, s) => (e(), t("col", {
                  key: s,
                  span: 1,
                  width: o.width
                }, null, 8, ae))), 128))
              ]),
              w.value ? a("", !0) : (e(), t("thead", {
                key: 0,
                align: l.align
              }, [
                d("tr", null, [
                  l.select ? (e(), t("th", re, [
                    m(i(f), {
                      modelValue: _.value,
                      "onUpdate:modelValue": n[1] || (n[1] = (o) => _.value = o),
                      disabled: !r.value.length
                    }, null, 8, ["modelValue", "disabled"])
                  ])) : a("", !0),
                  l.num ? (e(), t("th", ue, "#")) : a("", !0),
                  (e(!0), t(u, null, c(l.columns, (o, s) => (e(), t("th", { key: s }, [
                    i(C)(o.title) ? (e(), p(b(g(o.title, o, s)), { key: 0 })) : (e(), t(u, { key: 1 }, [
                      v(k(o.title), 1)
                    ], 64))
                  ]))), 128))
                ])
              ], 8, ne)),
              d("tbody", { align: l.align }, [
                (e(!0), t(u, null, c(r.value, (o, s) => (e(), t("tr", { key: s }, [
                  l.select ? (e(), t("td", ie, [
                    m(i(f), {
                      modelValue: o._select,
                      "onUpdate:modelValue": (y) => o._select = y,
                      "on-change": D
                    }, null, 8, ["modelValue", "onUpdate:modelValue"])
                  ])) : a("", !0),
                  l.num ? (e(), t("td", he, k(s + 1), 1)) : a("", !0),
                  (e(!0), t(u, null, c(l.columns, (y, H) => (e(), t("td", { key: H }, [
                    y.render ? (e(), p(b(N(y.render, o, y, s)), { key: 0 })) : (e(), t(u, { key: 1 }, [
                      y.key ? (e(), t(u, { key: 0 }, [
                        v(k(o[y.key]), 1)
                      ], 64)) : a("", !0)
                    ], 64))
                  ]))), 128))
                ]))), 128))
              ], 8, de),
              l.$slots.tfoot ? (e(), t("tfoot", ce, [
                V(l.$slots, "tfoot")
              ])) : a("", !0)
            ])) : (e(), t(u, { key: 1 }, [
              d("table", ye, [
                d("colgroup", null, [
                  l.select ? (e(), t("col", _e)) : a("", !0),
                  l.num ? (e(), t("col", me)) : a("", !0),
                  (e(!0), t(u, null, c(l.columns, (o, s) => (e(), t("col", {
                    key: s,
                    span: 1,
                    width: o.width
                  }, null, 8, ke))), 128))
                ]),
                w.value ? a("", !0) : (e(), t("thead", {
                  key: 0,
                  align: l.align
                }, [
                  d("tr", null, [
                    l.select ? (e(), t("th", be, [
                      m(i(f), {
                        modelValue: _.value,
                        "onUpdate:modelValue": n[2] || (n[2] = (o) => _.value = o),
                        disabled: !r.value.length
                      }, null, 8, ["modelValue", "disabled"])
                    ])) : a("", !0),
                    l.num ? (e(), t("th", ve, "#")) : a("", !0),
                    (e(!0), t(u, null, c(l.columns, (o, s) => (e(), t("th", { key: s }, [
                      i(C)(o.title) ? (e(), p(b(g(o.title, o, s)), { key: 0 })) : (e(), t(u, { key: 1 }, [
                        v(k(o.title), 1)
                      ], 64))
                    ]))), 128))
                  ])
                ], 8, pe))
              ]),
              d("div", fe, [
                V(l.$slots, "noData", {}, () => [
                  m(i(G), { content: "暂无数据" })
                ])
              ])
            ], 64))
          ])
        ], 64)) : (e(), t("table", ge, [
          V(l.$slots, "default")
        ]))
      ])
    ], 6));
  }
});
export {
  ze as default
};
