// fighting-design/es/_hooks/use-message-work/index.js
import { getCurrentInstance as I, ref as a, computed as m } from "vue";
import { removeInstance as L } from "../use-message/index.js";
import { useList as g } from "../use-list/index.js";
import { useRun as p } from "../use-run/index.js";
const M = (e, c) => {
  const { classes: u, styles: f } = g(e, c), { run: d } = p(), o = I(), l = a(!0), v = u(["type", "placement", "round"], `f-${c}`), T = f(["color", "background", "zIndex"], "zIndex"), y = m(() => e.placement.includes("top")), n = a(e.offset);
  let s;
  const r = (t) => {
    l.value = !1, d(e.onClose, t), clearTimeout(s);
  }, x = () => {
    L(o);
  }, C = () => {
    o.vnode.el && o.vnode.el.parentElement.removeChild(o.vnode.el);
  }, i = () => {
    e.duration > 0 && (s = setTimeout(() => {
      r();
    }, e.duration));
  };
  i();
  const b = m(() => {
    const t = {};
    return e.placement.includes("bottom") ? t.bottom = n.value + "px" : t.top = n.value + "px", t;
  });
  return {
    classList: v,
    style: T,
    visible: l,
    isPosition: y,
    offsetStyle: b,
    offsetVal: n,
    onBeforeLeave: x,
    onAfterLeave: C,
    clearTimer: () => {
      s && clearTimeout(s);
    },
    startTime: () => {
      i();
    },
    handelClose: r
  };
};
export {
  M as useMessageWork
};
