// fighting-design/es/image-preview/src/index.js
import { defineComponent as O, mergeModels as P, useModel as S, ref as V, openBlock as m, createBlock as _, Teleport as Z, createVNode as i, Transition as h, withCtx as f, withDirectives as b, createElementVNode as l, normalizeStyle as I, unref as e, withModifiers as A, vShow as k, createCommentVNode as C, createElementBlock as $, Fragment as D } from "vue";
import { Props as j } from "./index3.js";
import { FButton as p } from "../../button/index.js";
import { FToolbar as q } from "../../toolbar/index.js";
import { FToolbarItem as t } from "../../toolbar-item/index.js";
import { useOperationImg as G } from "../../_hooks/use-operation-img/index.js";
import H from "../../_svg/f-icon-zoom-out/index.js";
import J from "../../_svg/f-icon-zoom-in/index.js";
import K from "../../_svg/f-icon-layout-rows/index.js";
import Q from "../../_svg/f-icon-rotate-clockwise/index.js";
import U from "../../_svg/f-icon-rotate-anti-clockwise/index.js";
import W from "../../_svg/f-icon-chevron-right/index.js";
import X from "../../_svg/f-icon-chevron-left/index.js";
import Y from "../../_svg/f-icon-cross/index.js";
import { isArray as ee } from "../../_utils/is/index.js";
import { useRun as oe } from "../../_hooks/use-run/index.js";
const ie = /* @__PURE__ */ l("div", { class: "f-image-preview__mask" }, null, -1), re = { class: "f-image-preview__wrapper" }, ne = ["src"];
var Ie = /* @__PURE__ */ O({
  name: "FImagePreview",
  __name: "image-preview",
  props: /* @__PURE__ */ P(j, {
    visible: { type: Boolean, default: !1, type: Boolean },
    visibleModifiers: {}
  }),
  emits: ["update:visible"],
  setup(u) {
    const r = u, a = S(u, "visible"), { run: c } = oe(), {
      scale: F,
      rotate: y,
      smaller: x,
      bigger: L,
      scrollZoom: d,
      recovery: v,
      rotateClockwise: B,
      rotateCounterClock: M
    } = G(), n = V(
      r.showIndex > r.imgList.length - 1 ? 0 : r.showIndex
    ), R = () => {
      ee(r.imgList) && r.imgList.forEach((o) => {
        const s = new Image();
        s.src = o;
      });
    }, T = {
      /** 下一张切换 */
      next: () => {
        if (n.value < r.imgList.length - 1) {
          n.value++;
          return;
        }
        n.value = 0;
      },
      /** 上一张切换 */
      prev: () => {
        if (n.value > 0) {
          n.value--;
          return;
        }
        n.value = r.imgList.length - 1;
      }
    }, g = (o) => {
      v(), c(T[o]);
    }, E = {
      1: () => x(),
      2: () => L(),
      3: () => v(),
      4: () => B(),
      5: () => M()
    }, z = (o) => {
      o && c(E[o]);
    }, w = (o) => {
      a.value = !1, c(r.onClose, o);
    };
    return (o, s) => (m(), _(Z, {
      to: "body",
      disabled: !o.appendToBody
    }, [
      i(h, {
        name: "f-image-preview__transition",
        onBeforeEnter: R
      }, {
        default: f(() => [
          b(l("div", {
            class: "f-image-preview",
            style: I({ zIndex: o.zIndex }),
            onMousewheel: s[0] || (s[0] = //@ts-ignore
            (...N) => e(d) && e(d)(...N))
          }, [
            ie,
            l("div", {
              class: "f-image-preview__container",
              onClick: A(w, ["self"])
            }, [
              i(h, { name: "f-image-preview__wrapper-transition" }, {
                default: f(() => [
                  b(l("div", re, [
                    l("img", {
                      class: "f-image-preview__node",
                      draggable: "false",
                      src: o.imgList[n.value],
                      style: I({
                        transform: `scale(${e(F)}) rotate(${e(y)}deg)`,
                        borderRadius: o.round
                      })
                    }, null, 12, ne)
                  ], 512), [
                    [k, a.value]
                  ])
                ]),
                _: 1
              }),
              o.isOption ? (m(), _(e(q), {
                key: 0,
                class: "f-image-preview__option",
                round: "",
                "on-click": z
              }, {
                default: f(() => [
                  i(e(t), {
                    icon: e(H),
                    index: 1
                  }, null, 8, ["icon"]),
                  i(e(t), {
                    icon: e(J),
                    index: 2
                  }, null, 8, ["icon"]),
                  i(e(t), {
                    icon: e(K),
                    index: 3
                  }, null, 8, ["icon"]),
                  i(e(t), {
                    icon: e(Q),
                    index: 4
                  }, null, 8, ["icon"]),
                  i(e(t), {
                    icon: e(U),
                    index: 5
                  }, null, 8, ["icon"])
                ]),
                _: 1
              })) : C("", !0),
              o.imgList.length > 1 ? (m(), $(D, { key: 1 }, [
                i(e(p), {
                  class: "f-image-preview__next",
                  circle: "",
                  "before-icon": e(W),
                  "on-click": () => g("next")
                }, null, 8, ["before-icon", "on-click"]),
                i(e(p), {
                  class: "f-image-preview__prev",
                  circle: "",
                  "before-icon": e(X),
                  "on-click": () => g("prev")
                }, null, 8, ["before-icon", "on-click"])
              ], 64)) : C("", !0),
              i(e(p), {
                class: "f-image-preview__close",
                circle: "",
                "before-icon": e(Y),
                "on-click": w
              }, null, 8, ["before-icon"])
            ])
          ], 36), [
            [k, a.value]
          ])
        ]),
        _: 1
      })
    ], 8, ["disabled"]));
  }
});
export {
  Ie as default
};
