// fighting-design/es/alert/src/index.js
import { defineComponent as g, ref as u, onMounted as I, openBlock as t, createBlock as a, Transition as S, withCtx as f, createElementBlock as r, normalizeClass as $, unref as s, normalizeStyle as z, renderSlot as n, createCommentVNode as l, createElementVNode as B, createTextVNode as E, toDisplayString as m, Fragment as F, renderList as N } from "vue";
import { Props as M } from "./index3.js";
import { FCloseBtn as V } from "../../close-btn/index.js";
import { FSvgIcon as w } from "../../svg-icon/index.js";
import { useAlertList as A } from "../../_hooks/use-alert-list/index.js";
import { useGlobal as P } from "../../_hooks/use-global/index.js";
import { useList as T } from "../../_hooks/use-list/index.js";
import { isNumber as D } from "../../_utils/is/index.js";
import { useRun as G } from "../../_hooks/use-run/index.js";
const R = { class: "f-alert__content" }, j = {
  key: 0,
  class: "f-alert__title"
}, q = {
  key: 1,
  class: "f-alert__sub-title"
};
var x = /* @__PURE__ */ g({
  name: "FAlert",
  __name: "alert",
  props: M,
  setup(d) {
    const o = d, { getProp: p } = P(o), { run: y } = G(), { styles: v, classes: k } = T(p(["type"]), "alert"), c = u(!0), i = u(), L = k(
      ["type", "bold", "simple", "center", "round", "fixed"],
      "f-alert"
    ), _ = v(["fontSize", "color", "background", "titleSize", "titleColor"]), b = (e) => {
      c.value = !1, y(o.onClose, e);
    };
    return I(() => {
      o.alertList && o.alertList.length && i.value && A(i.value)(D(o.duration) ? o.duration : 2e3);
    }), (e, H) => (t(), a(S, { name: "f-alert" }, {
      default: f(() => [
        c.value ? (t(), r("div", {
          key: 0,
          role: "alert",
          class: $(s(L)),
          style: z(s(_))
        }, [
          e.$slots.beforeIcon || e.beforeIcon ? (t(), a(s(w), {
            key: 0,
            icon: e.beforeIcon
          }, {
            default: f(() => [
              n(e.$slots, "beforeIcon")
            ]),
            _: 3
          }, 8, ["icon"])) : l("", !0),
          B("div", R, [
            e.title ? (t(), r("div", j, [
              n(e.$slots, "title", {}, () => [
                E(m(e.title), 1)
              ])
            ])) : l("", !0),
            e.$slots.default ? (t(), r("div", q, [
              n(e.$slots, "default")
            ])) : l("", !0),
            e.alertList && e.alertList.length ? (t(), r("div", {
              key: 2,
              ref_key: "alertListEl",
              ref: i,
              class: "f-alert__list"
            }, [
              (t(!0), r(F, null, N(e.alertList, (h, C) => (t(), r("div", {
                key: C,
                class: "f-alert__list-item"
              }, m(h), 1))), 128))
            ], 512)) : l("", !0)
          ]),
          e.close ? (t(), a(s(V), {
            key: 1,
            round: "",
            color: e.type === "default" ? "#333" : "#fff",
            size: 16,
            icon: e.closeIcon,
            "on-click": b
          }, {
            default: f(() => [
              n(e.$slots, "closeIcon")
            ]),
            _: 3
          }, 8, ["color", "icon"])) : l("", !0)
        ], 6)) : l("", !0)
      ]),
      _: 3
    }));
  }
});
export {
  x as default
};
