// fighting-design/es/alert/src/index3.js
import { setStringProp as e, setStringNumberProp as r, setBooleanProp as o, setObjectProp as t, setArrayProp as n, setNumberProp as s, setFunctionProp as l } from "../../_utils/props/index.js";
import { FIGHTING_TYPE as p } from "../../_tokens/attrs/index.js";
const u = {
  /**
   * 类型
   *
   * @values default primary success danger warning info
   * @default undefined
   */
  type: e(void 0, (i) => p.includes(i)),
  /** 文字大小 */
  fontSize: r(),
  /** title 大小 */
  titleSize: r(),
  /** 是否加粗 */
  bold: o(),
  /** 是否居中 */
  center: o(),
  /** 是否可关闭 */
  close: o(),
  /** 是否为简约的 */
  simple: o(),
  /** 自定义标题 */
  title: e(),
  /** 是否为圆角 */
  round: o(),
  /** 自定义背景色 */
  background: e(),
  /** 自定义颜色 */
  color: e(),
  /** 自定义 title 颜色 */
  titleColor: e(),
  /** 是否为固定定位 */
  fixed: o(),
  /** 自定义关闭 icon */
  closeIcon: t(),
  /** 自定义之前的 icon */
  beforeIcon: t(),
  /** 滚动列表 */
  alertList: n([]),
  /** 滚动列表动画间隔 */
  duration: s(),
  /** 点击关闭之后执行的回调 */
  onClose: l()
};
export {
  u as Props
};
