// fighting-design/es/_hooks/use-loading/index.js
import { createApp as c } from "vue";
import u from "../../loading/src/index.js";
const g = () => {
  const r = (o, t) => {
    const e = (n) => t.value[n], i = (n) => e(n) || o.getAttribute(`f-loading-${n}`) || "";
    return {
      visible: !!t.value,
      fullscreen: t.modifiers.fullscreen,
      text: i("text"),
      color: i("color"),
      background: i("background")
    };
  }, a = (o, t) => {
    o.originalPosition !== "absolute" && o.originalPosition !== "fixed" && (o.style.position = "relative");
    const e = r(o, t), i = c(u, e), n = i.mount(
      document.createElement("div")
    );
    o.vm = n, o.loadingInstance = i, o.appendChild(n.$el);
  }, s = (o) => {
    o.loadingInstance && (o.style.position = o.originalPosition, o.removeChild(o.vm.$el), o.loadingInstance.unmount(), o.loadingInstance = null);
  };
  return {
    /**
     * 在绑定元素的父组件
     *
     * 及他自己的所有子节点都挂载完成后调用
     *
     * @param { Object } el 指令绑定到的元素。这可以用于直接操作 DOM
     * @param { Object } binding 一个对象，包含一些配置参数
     */
    mounted: (o, t) => {
      const e = getComputedStyle(o).position || "static";
      o.originalPosition = e, t.value && a(o, t);
    },
    /**
     * 在绑定元素的父组件
     *
     * 及他自己的所有子节点都更新后调用
     *
     * @param { Object } el 指令绑定到的元素。这可以用于直接操作 DOM
     * @param { Object } binding 一个对象，包含一些配置参数
     */
    updated: (o, t) => {
      t.value !== t.oldValue && (t.value ? a(o, t) : s(o));
    }
  };
};
export {
  g as useLoading
};
