// fighting-design/es/page-header/src/index.js
import { defineComponent as h, computed as g, openBlock as o, createElementBlock as s, createElementVNode as t, createVNode as _, unref as l, withCtx as v, renderSlot as a, createBlock as C, resolveDynamicComponent as b, toDisplayString as r, normalizeClass as n, normalizeStyle as k, createTextVNode as i, createCommentVNode as y } from "vue";
import { Props as z } from "./index3.js";
import { FSvgIcon as B } from "../../svg-icon/index.js";
import S from "../../_svg/f-icon-arrow-left/index.js";
import { useGlobal as N } from "../../_hooks/use-global/index.js";
import { useRun as V } from "../../_hooks/use-run/index.js";
const $ = { class: "f-page-header" }, F = { class: "f-page-header__text" }, w = {
  key: 0,
  class: "f-page-header__main-subtitle"
};
var A = /* @__PURE__ */ h({
  name: "FPageHeader",
  __name: "page-header",
  props: z,
  setup(d) {
    const m = d, { run: c } = V(), { getLang: p } = N(), f = g(() => p("pageHeader").value.backText), u = (e) => {
      c(m.onBack, e);
    };
    return (e, D) => (o(), s("header", $, [
      t("div", {
        class: "f-page-header__left",
        onClick: u
      }, [
        _(l(B), { size: e.iconSize }, {
          default: v(() => [
            a(e.$slots, "icon", {}, () => [
              (o(), C(b(e.icon || l(S))))
            ])
          ]),
          _: 3
        }, 8, ["size"]),
        t("div", F, r(f.value), 1)
      ]),
      t("div", {
        class: n(["f-page-header__main", { "f-page-header__main-center": e.titleCenter }])
      }, [
        t("div", {
          style: k({ color: e.titleColor }),
          class: n([
            "f-page-header__main-title",
            { "f-page-header__main-title-bold": e.titleBold }
          ])
        }, [
          a(e.$slots, "title", {}, () => [
            i(r(e.title), 1)
          ])
        ], 6),
        e.subtitle || e.$slots.subtitle ? (o(), s("div", w, [
          a(e.$slots, "subtitle", {}, () => [
            i(r(e.subtitle), 1)
          ])
        ])) : y("", !0)
      ], 2)
    ]));
  }
});
export {
  A as default
};
