// fighting-design/es/calendar/src/index.js
import { defineComponent as j, computed as y, reactive as q, watch as D, onUnmounted as A, resolveComponent as k, openBlock as r, createElementBlock as l, normalizeClass as F, unref as c, normalizeStyle as J, createVNode as s, createElementVNode as f, withCtx as d, Fragment as h, renderList as _, createTextVNode as w, toDisplayString as u, createCommentVNode as L } from "vue";
import { Props as K } from "./index3.js";
import { FSvgIcon as M } from "../../svg-icon/index.js";
import { isDate as O } from "../../_utils/is/index.js";
import Q from "../../_svg/f-icon-chevron-left/index.js";
import W from "../../_svg/f-icon-chevron-right/index.js";
import { useCalendar as X } from "../../_hooks/use-calendar/index.js";
import { useGlobal as Z } from "../../_hooks/use-global/index.js";
import { useRun as ee } from "../../_hooks/use-run/index.js";
import { useList as te } from "../../_hooks/use-list/index.js";
const ne = {
  key: 0,
  class: "f-calendar__header"
}, oe = { class: "f-calendar__option" }, ae = { class: "f-calendar__week" }, re = { class: "f-calendar__day" }, le = ["onClick"], se = { class: "f-calendar__solar" }, ce = {
  key: 0,
  class: "f-calendar__lunar"
};
var ye = /* @__PURE__ */ j({
  name: "FCalendar",
  __name: "calendar",
  props: K,
  setup(x) {
    const a = x, { getLang: b } = Z(), { run: p } = ee(), { styles: N, classes: $ } = te(a, "calendar"), { generateCalendar: H } = X(a), v = y(() => O(a.date) ? a.date : /* @__PURE__ */ new Date()), e = q({
      year: v.value.getFullYear(),
      month: v.value.getMonth() + 1,
      date: v.value.getDate()
    }), I = y(() => H(e.year, e.month)), S = y(() => b("calendar").value.weekList), U = () => {
      e.month === 1 ? (e.year -= 1, e.month = 12) : e.month -= 1;
    }, z = () => {
      e.month === 12 ? (e.year += 1, e.month = 1) : e.month += 1;
    }, i = {
      prev: () => U(),
      next: () => z(),
      current: () => {
        e.year = a.date.getFullYear(), e.month = a.date.getMonth() + 1, e.date = a.date.getDate();
      }
    }, C = (o) => {
      i[o] && i[o]();
    }, B = (o) => {
      const { target: n, day: m } = o;
      n === "prev" && i[n](), n === "next" && i[n](), n === "current" && (e.date = m, p(a.onChangeDate, e.year, e.month, e.date));
    }, E = D(
      () => e.month,
      /**
       * @param { number } month 最新的月份
       */
      (o) => {
        p(a.onChangeMonth, e.year, o, e.date);
      }
    ), R = D(
      () => e,
      () => {
        p(a.onChange, e.year, e.month, e.date);
      },
      { deep: !0 }
    ), Y = (o, n) => n === e.date && o === e.month ? "f-calendar__day-today" : "";
    A(() => {
      R(), E();
    });
    const G = N(["borderColor", "dayCellHeight", "weekCellHeight"]), P = $(["border"], "f-calendar");
    return (o, n) => {
      const m = k("f-option"), V = k("f-select"), T = k("f-button");
      return r(), l("div", {
        class: F(c(P)),
        style: J(c(G))
      }, [
        o.showHeader ? (r(), l("header", ne, [
          s(c(M), {
            icon: c(Q),
            onClick: n[0] || (n[0] = (t) => C("prev"))
          }, null, 8, ["icon"]),
          f("div", oe, [
            s(V, {
              modelValue: e.year,
              "onUpdate:modelValue": n[1] || (n[1] = (t) => e.year = t),
              width: 116
            }, {
              default: d(() => [
                (r(), l(h, null, _(200, (t) => s(m, {
                  key: t,
                  value: 1900 + t
                }, {
                  default: d(() => [
                    w(u(1900 + t), 1)
                  ]),
                  _: 2
                }, 1032, ["value"])), 64))
              ]),
              _: 1
            }, 8, ["modelValue"]),
            s(V, {
              modelValue: e.month,
              "onUpdate:modelValue": n[2] || (n[2] = (t) => e.month = t),
              width: 116
            }, {
              default: d(() => [
                (r(), l(h, null, _(12, (t) => s(m, {
                  key: t,
                  value: t
                }, {
                  default: d(() => [
                    w(u(t), 1)
                  ]),
                  _: 2
                }, 1032, ["value"])), 64))
              ]),
              _: 1
            }, 8, ["modelValue"]),
            s(T, {
              type: "default",
              onClick: n[3] || (n[3] = (t) => C("current"))
            }, {
              default: d(() => [
                w("今天")
              ]),
              _: 1
            })
          ]),
          s(c(M), {
            icon: c(W),
            onClick: n[4] || (n[4] = (t) => C("next"))
          }, null, 8, ["icon"])
        ])) : L("", !0),
        f("div", ae, [
          (r(!0), l(h, null, _(S.value, (t, g) => (r(), l("div", {
            key: g,
            class: "f-calendar__week-item"
          }, u(t), 1))), 128))
        ]),
        f("div", re, [
          (r(!0), l(h, null, _(I.value, (t, g) => (r(), l("div", {
            key: g,
            class: F([
              "f-calendar__day-item",
              {
                "f-calendar__day-current": t.target === "current",
                [Y(t.month, t.day)]: t.target === "current"
              }
            ]),
            onClick: (de) => B(t)
          }, [
            f("span", se, u(t.day), 1),
            o.lunar ? (r(), l("span", ce, u(t.lunarFestival || t.festival || t.term || t.lunar), 1)) : L("", !0)
          ], 10, le))), 128))
        ])
      ], 6);
    };
  }
});
export {
  ye as default
};
