// fighting-design/es/tabs/src/index3.js
import { setStringNumberProp as r, setStringProp as t, setFunctionProp as n } from "../../_utils/props/index.js";
const i = {
  /** 活跃的 name */
  modelValue: r(),
  /**
   * 头部位置
   *
   * @values left right top bottom
   * @default top
   */
  position: t("top", (e) => ["left", "right", "top", "bottom"].includes(e)),
  /**
   * 样式风格
   *
   * @values line card segment
   * @default line
   */
  type: t("line", (e) => ["line", "card", "segment"].includes(e)),
  /**
   * 对齐方式
   *
   * @values flex-start center flex-end space-between space-around space-evenly
   * @see justify-content https://developer.mozilla.org/zh-CN/docs/Web/CSS/justify-content
   */
  justifyContent: t(
    void 0,
    (e) => [
      "flex-start",
      "center",
      "flex-end",
      "space-between",
      "space-around",
      "space-evenly"
    ].includes(e)
  ),
  /**
   * 触发方式
   *
   * @values click hover
   * @default click
   */
  trigger: t("click", (e) => ["click", "hover"].includes(e)),
  /** 切换前的回调 */
  onSwitch: n()
}, s = Symbol("tabs-props-key");
export {
  i as Props,
  s as TABS_PROPS_KEY
};
