// fighting-design/es/list/src/index.js
import { defineComponent as d, provide as p, openBlock as s, createElementBlock as o, normalizeClass as _, unref as l, normalizeStyle as u, renderSlot as t, createCommentVNode as a } from "vue";
import { Props as h, LIST_PROPS_KEY as y } from "./index3.js";
import { useList as z } from "../../_hooks/use-list/index.js";
const C = {
  key: 0,
  class: "f-list__header"
}, $ = {
  key: 1,
  class: "f-list__footer"
};
var P = /* @__PURE__ */ d({
  name: "FList",
  __name: "list",
  props: h,
  setup(n) {
    const r = n;
    p(y, r);
    const { classes: i, styles: m } = z(r, "list"), c = m(["borderColor", "maxHeight", "zebraColor"]), f = i(["zebra", "center", "size"], "f-list");
    return (e, k) => (s(), o("ul", {
      role: "list",
      class: _(l(f)),
      style: u(l(c))
    }, [
      e.$slots.header ? (s(), o("header", C, [
        t(e.$slots, "header")
      ])) : a("", !0),
      t(e.$slots, "default"),
      e.$slots.footer ? (s(), o("footer", $, [
        t(e.$slots, "footer")
      ])) : a("", !0)
    ], 6));
  }
});
export {
  P as default
};
