// fighting-design/es/link/src/index3.js
import { setStringProp as o, setStringNumberProp as n, setBooleanProp as e, setObjectProp as t, setFunctionProp as i } from "../../_utils/props/index.js";
import { FIGHTING_TYPE as s, FIGHTING_TARGET as c } from "../../_tokens/attrs/index.js";
const u = {
  /**
   * link 的类型
   *
   * @values default primary success danger warning
   */
  type: o(void 0, (r) => s.includes(r)),
  /**
   * 链接的地址
   *
   * @see href https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/a?qs=%3Ca%3E#attr-href
   */
  href: o(),
  /** 的文字大小 */
  size: n(),
  /**
   * 状态的样式状态
   *
   * @values line background
   */
  state: o(void 0, (r) => ["line", "background"].includes(r)),
  /** 是否禁用 */
  disabled: e(),
  /**
   * 原生 target 属性
   *
   * @values _blank _self _parent _top
   * @default _self
   * @see target https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/a#attr-target
   */
  target: o("_self", (r) => c.includes(r)),
  /** 自定义颜色 */
  color: o(),
  /** 自定义移入状态颜色 */
  hoverColor: o(),
  /** 是否禁止 copy */
  noCopy: e(),
  /** 是否禁止链接跳转及重新加载（取消默认事件） */
  noLink: e(),
  /** 之前的 icon */
  beforeIcon: t(),
  /** 之后的 icon */
  afterIcon: t(),
  /** 点击执行的回调 */
  onClick: i()
};
export {
  u as Props
};
