// fighting-design/es/avatar-group/src/index.js
import { defineComponent as p, provide as f, ref as u, onMounted as c, openBlock as i, createElementBlock as v, normalizeStyle as m, unref as d, renderSlot as y } from "vue";
import { Props as _, AVATAR_GROUP_PROPS_KEY as S } from "./index3.js";
import { useList as g } from "../../_hooks/use-list/index.js";
var G = /* @__PURE__ */ p({
  name: "FAvatarGroup",
  __name: "avatar-group",
  props: _,
  setup(a) {
    const r = a, { styles: n } = g(r, "avatar-group");
    f(S, r);
    const t = u();
    c(() => {
      if (t.value) {
        const e = t.value.querySelectorAll(
          ".f-avatar"
        );
        e && e.length && e.forEach((o, l) => {
          o.style.zIndex = (e.length - l).toString();
        });
      }
    });
    const s = n(["positionSize"]);
    return (e, o) => (i(), v("div", {
      ref_key: "avatarGroupRef",
      ref: t,
      role: "group",
      class: "f-avatar-group",
      style: m(d(s))
    }, [
      y(e.$slots, "default")
    ], 4));
  }
});
export {
  G as default
};
