// fighting-design/es/notification/src/index.js
import { defineComponent as b, computed as d, openBlock as t, createBlock as a, Transition as T, unref as e, withCtx as w, withDirectives as z, createElementVNode as v, normalizeClass as A, normalizeStyle as D, createElementBlock as n, createVNode as M, createCommentVNode as l, Fragment as y, isVNode as g, resolveDynamicComponent as k, toDisplayString as m, createTextVNode as E, vShow as W } from "vue";
import { Props as P } from "./index3.js";
import { FSvgIcon as R } from "../../svg-icon/index.js";
import { FCloseBtn as U } from "../../close-btn/index.js";
import j from "../../_svg/f-icon-smile-line/index.js";
import q from "../../_svg/f-icon-lightbulb/index.js";
import G from "../../_svg/f-icon-thumb-up/index.js";
import H from "../../_svg/f-icon-circle-cross/index.js";
import J from "../../_svg/f-icon-warning/index.js";
import K from "../../_svg/f-icon-bell/index.js";
import { useMessageWork as O } from "../../_hooks/use-message-work/index.js";
import { isString as Q } from "../../_utils/is/index.js";
const X = {
  key: 0,
  class: "f-notification__icon"
}, Y = { class: "f-notification__content" }, Z = {
  key: 1,
  class: "f-notification__title"
}, $ = {
  key: 2,
  class: "f-notification__message"
};
var fe = /* @__PURE__ */ b({
  name: "FNotification",
  __name: "notification",
  props: P,
  setup(B, { expose: C }) {
    const s = B, {
      classList: F,
      styleList: L,
      visible: h,
      offsetStyle: I,
      offsetVal: _,
      onBeforeLeave: S,
      onAfterLeave: N,
      clearTimer: c,
      startTime: f,
      handelClose: p
    } = O(s, "notification"), u = d(() => s.icon ? s.icon : {
      default: j,
      primary: q,
      success: G,
      danger: H,
      warning: J,
      info: K
    }[s.type]), V = d(() => s.placement.includes("right"));
    return C({ offsetVal: _ }), (o, i) => (t(), a(T, {
      mode: "out-in",
      appear: "",
      name: "f-notification-fade" + (V.value ? "-right" : "-left"),
      onBeforeLeave: e(S),
      onAfterLeave: e(N)
    }, {
      default: w(() => [
        z(v("div", {
          class: A(e(F)),
          style: D([e(I), e(L)]),
          onMouseleave: i[1] || (i[1] = //@ts-ignore
          (...r) => e(f) && e(f)(...r)),
          onMouseenter: i[2] || (i[2] = //@ts-ignore
          (...r) => e(c) && e(c)(...r))
        }, [
          u.value ? (t(), n("div", X, [
            M(e(R), {
              icon: u.value,
              size: 25
            }, null, 8, ["icon"])
          ])) : l("", !0),
          v("div", Y, [
            o.title ? (t(), n(y, { key: 0 }, [
              g(o.title) ? (t(), a(k(o.title), { key: 0 })) : (t(), n("h3", Z, m(o.title), 1))
            ], 64)) : l("", !0),
            g(o.message) ? (t(), a(k(o.message), { key: 1 })) : (t(), n("div", $, m(o.message), 1))
          ]),
          o.close ? (t(), n("div", {
            key: 1,
            class: "f-notification__close",
            onClick: i[0] || (i[0] = //@ts-ignore
            (...r) => e(p) && e(p)(...r))
          }, [
            e(Q)(o.closeBtn) ? (t(), n(y, { key: 0 }, [
              E(m(o.closeBtn), 1)
            ], 64)) : (t(), a(e(U), {
              key: 1,
              icon: o.closeBtn,
              size: 15,
              color: "#a4a4a4"
            }, null, 8, ["icon"]))
          ])) : l("", !0)
        ], 38), [
          [W, e(h)]
        ])
      ]),
      _: 1
    }, 8, ["name", "onBeforeLeave", "onAfterLeave"]));
  }
});
export {
  fe as default
};
