// fighting-design/es/notification/src/index3.js
import { setStringProp as t, setBooleanProp as o, setNumberProp as r, setObjectProp as n, setFunctionProp as l } from "../../_utils/props/index.js";
import { FIGHTING_TYPE as p } from "../../_tokens/attrs/index.js";
const u = {
  /** 通知标题 */
  title: {
    type: [String, Object],
    default: () => null
  },
  /** 通知文本 */
  message: {
    type: [String, Object],
    default: () => null
  },
  /**
   * 消息类型
   *
   * @values default primary success danger warning
   */
  type: t(void 0, (e) => p.includes(e)),
  /** 是否可关闭 */
  close: o(),
  /** 显示时间，单位为毫秒。 设为 0 则不会自动关闭 */
  duration: r(2500),
  /** 是否为圆角类型 */
  round: o(),
  /** 是否显示按钮 */
  showIcon: o(!0),
  /** 自定义前缀 icon */
  icon: n(),
  /** 自定义字体颜色 */
  color: t(),
  /** 自定义背景色 */
  background: t(),
  /** 偏移距离 */
  offset: r(20),
  /**
   * 弹出位置
   *
   * @values top-left top-right bottom-left bottom-right
   * @default top-right
   */
  placement: t(
    "top-right",
    (e) => ["top-left", "top-right", "bottom-left", "bottom-right"].includes(
      e
    )
  ),
  /** 层级 */
  zIndex: r(),
  /** 自定义关闭按钮 */
  closeBtn: {
    type: [String, Object],
    default: () => null
  },
  /** 关闭之后回调 */
  onClose: l()
};
export {
  u as Props
};
