// fighting-design/es/toolbar-item/src/index.js
import { defineComponent as a, openBlock as n, createElementBlock as p, normalizeProps as u, unref as e, createBlock as f, createCommentVNode as d, renderSlot as k } from "vue";
import { Props as C } from "./index3.js";
import { FSvgIcon as E } from "../../svg-icon/index.js";
import { TOOLBAR_ITEM_CLASS_NAME as _, TOOLBAR_INDEX as A } from "../../_tokens/key/index3.js";
import { useList as B } from "../../_hooks/use-list/index.js";
import { useRun as S } from "../../_hooks/use-run/index.js";
var N = /* @__PURE__ */ a({
  name: "FToolbarItem",
  __name: "toolbar-item",
  props: C,
  setup(t) {
    const r = t, { run: s } = S(), { styles: i } = B(r, "toolbar-item"), l = (o) => {
      s(r.onClick, o);
    }, m = i(["color"]), c = A;
    return (o, T) => (n(), p("span", u({
      class: e(_),
      [e(c) || ""]: o.index,
      style: e(m),
      onClick: l
    }), [
      o.icon ? (n(), f(e(E), {
        key: 0,
        size: o.iconSize,
        icon: o.icon
      }, null, 8, ["size", "icon"])) : d("", !0),
      k(o.$slots, "default")
    ], 16));
  }
});
export {
  N as default
};
