// fighting-design/es/message/src/index.js
import { defineComponent as V, openBlock as s, createBlock as a, Transition as N, unref as e, withCtx as z, withDirectives as F, createElementVNode as M, normalizeClass as T, normalizeStyle as b, createElementBlock as r, createVNode as w, createCommentVNode as f, isVNode as A, resolveDynamicComponent as D, toDisplayString as p, Fragment as _, createTextVNode as E, vShow as P } from "vue";
import { Props as h } from "./index3.js";
import { FSvgIcon as I } from "../../svg-icon/index.js";
import { FCloseBtn as W } from "../../close-btn/index.js";
import { useMessageWork as j } from "../../_hooks/use-message-work/index.js";
import { isString as q } from "../../_utils/is/index.js";
const G = {
  key: 0,
  class: "f-message__before_icon"
}, H = {
  key: 2,
  class: "f-message__text"
};
var X = /* @__PURE__ */ V({
  name: "FMessage",
  __name: "message",
  props: h,
  setup(c, { expose: d }) {
    const v = c, {
      classList: g,
      style: u,
      visible: y,
      isPosition: k,
      offsetStyle: B,
      offsetVal: C,
      onBeforeLeave: L,
      onAfterLeave: S,
      clearTimer: i,
      startTime: m,
      handelClose: l
    } = j(v, "message");
    return d({ offsetVal: C }), (o, t) => (s(), a(N, {
      mode: "out-in",
      appear: "",
      name: "f-message-fade" + (e(k) ? "-top" : "-bottom"),
      onBeforeLeave: e(L),
      onAfterLeave: e(S)
    }, {
      default: z(() => [
        F(M("div", {
          class: T(e(g)),
          style: b([e(B), e(u)]),
          onMouseleave: t[1] || (t[1] = //@ts-ignore
          (...n) => e(m) && e(m)(...n)),
          onMouseenter: t[2] || (t[2] = //@ts-ignore
          (...n) => e(i) && e(i)(...n))
        }, [
          o.icon ? (s(), r("div", G, [
            w(e(I), {
              icon: o.icon,
              size: 16
            }, null, 8, ["icon"])
          ])) : f("", !0),
          A(o.message) ? (s(), a(D(o.message), { key: 1 })) : (s(), r("div", H, p(o.message), 1)),
          o.close ? (s(), r("div", {
            key: 3,
            class: "f-message__close",
            onClick: t[0] || (t[0] = //@ts-ignore
            (...n) => e(l) && e(l)(...n))
          }, [
            e(q)(o.closeBtn) ? (s(), r(_, { key: 0 }, [
              E(p(o.closeBtn), 1)
            ], 64)) : (s(), a(e(W), {
              key: 1,
              icon: o.closeBtn,
              size: 15,
              color: "#a4a4a4"
            }, null, 8, ["icon"]))
          ])) : f("", !0)
        ], 38), [
          [P, e(y)]
        ])
      ]),
      _: 1
    }, 8, ["name", "onBeforeLeave", "onAfterLeave"]));
  }
});
export {
  X as default
};
