// fighting-design/es/message/src/index3.js
import { setStringProp as t, setBooleanProp as r, setNumberProp as e, setObjectProp as n, setFunctionProp as l } from "../../_utils/props/index.js";
import { FIGHTING_TYPE as p } from "../../_tokens/attrs/index.js";
const c = {
  /** 消息文本 */
  message: {
    type: [String, Object],
    default: () => null
  },
  /**
   * 消息类型
   *
   * @values default primary success danger warning
   * @default default
   */
  type: t("default", (o) => p.includes(o)),
  /** 是否可关闭 */
  close: r(),
  /** 显示时间，单位为毫秒。 设为 0 则不会自动关闭 */
  duration: e(2500),
  /** 是否为圆角类型 */
  round: r(),
  /** 消息 icon */
  icon: n(),
  /** 字体颜色 */
  color: t(),
  /** 自定义背景色 */
  background: t(),
  /** 偏移距离 */
  offset: e(20),
  /**
   * 位置
   *
   * @values top top-left top-right bottom bottom-left bottom-right
   * @default top
   */
  placement: t("top", (o) => ["top", "top-left", "top-right", "bottom", "bottom-left", "bottom-right"].includes(o)),
  /** 层级 */
  zIndex: e(),
  /** 关闭按钮 */
  closeBtn: {
    type: [String, Object],
    default: () => null
  },
  /** 关闭回调 */
  onClose: l()
};
export {
  c as Props
};
