// fighting-design/es/_hooks/use-global/index.js
import { inject as G, computed as o, reactive as a, toRefs as y } from "vue";
import { FIGHTING_GLOBAL_PROPS_KEY as z } from "../../fighting-global/src/index.js";
import { LANG as I } from "../../_lang/index.js";
import { FIGHTING_TYPE as L, FIGHTING_SIZE as P } from "../../_tokens/attrs/index.js";
import { isArray as N } from "../../_utils/is/index.js";
const S = (e) => {
  const r = G(z, null), u = (t = "default") => o(() => !e || e.type && !L.includes(e.type) ? t : e.type || r && r.type || t), c = (t = "middle", n) => o(() => !e || e.size && !P.includes(e.size) ? t : e.size || n || r && r.size || t), f = (t) => o(() => {
    const n = r && r.lang || "zh-CN";
    return I[n][t];
  }), l = {
    type: u,
    size: c
  };
  return {
    getType: u,
    getSize: c,
    getLang: f,
    getProp: (t, n) => {
      const i = {};
      return N(t) && t.forEach((s, g) => {
        if (l[s]) {
          const m = n && n[g];
          i[s] = l[s](m);
        }
      }), e ? a({
        ...y(e),
        ...i
      }) : a({
        ...i
      });
    }
  };
};
export {
  S as useGlobal
};
