// fighting-design/es/loading/src/index.js
import { defineComponent as p, openBlock as t, createElementBlock as n, normalizeClass as f, unref as e, normalizeStyle as u, createVNode as s, withCtx as g, renderSlot as _, toDisplayString as y, createCommentVNode as i } from "vue";
import { Props as k } from "./index3.js";
import { FSvgIcon as v } from "../../svg-icon/index.js";
import C from "../../_svg/f-icon-loading-a/index.js";
import { useList as L } from "../../_hooks/use-list/index.js";
const S = {
  key: 0,
  class: "f-loading__title"
};
var N = /* @__PURE__ */ p({
  name: "FLoading",
  __name: "loading",
  props: k,
  setup(r) {
    const a = r, { styles: l, classes: c } = L(a, "loading"), m = l(["background", "color"]), d = c(["fullscreen"], "f-loading");
    return (o, h) => o.visible ? (t(), n("div", {
      key: 0,
      class: f(e(d)),
      style: u(e(m))
    }, [
      s(e(v), {
        size: 20,
        class: "f-loading__animation",
        icon: o.icon
      }, {
        default: g(() => [
          _(o.$slots, "icon", {}, () => [
            s(e(C))
          ])
        ]),
        _: 3
      }, 8, ["icon"]),
      o.text ? (t(), n("span", S, y(o.text), 1)) : i("", !0)
    ], 6)) : i("", !0);
  }
});
export {
  N as default
};
