// fighting-design/es/text/src/index3.js
import { setStringProp as o, setStringNumberProp as e, setBooleanProp as r } from "../../_utils/props/index.js";
import { FIGHTING_TYPE as n } from "../../_tokens/attrs/index.js";
const l = {
  /**
   * 类型
   *
   * @values default primary success danger warning
   */
  type: o(void 0, (i) => n.includes(i)),
  /**
   * 文字大小
   *
   * @see font-size https://developer.mozilla.org/zh-CN/docs/Web/CSS/font-size
   */
  size: e(),
  /**
   * 文字颜色
   *
   * @see color https://developer.mozilla.org/zh-CN/docs/Web/CSS/color
   */
  color: o(),
  /**
   * 背景颜色
   *
   * @see background https://developer.mozilla.org/zh-CN/docs/Web/CSS/background
   */
  background: o(),
  /**
   * 是否独占一行
   *
   * @see display https://developer.mozilla.org/zh-CN/docs/Web/CSS/display
   */
  block: r(),
  /**
   * 文字间距
   *
   * @see letter-spacing https://developer.mozilla.org/zh-CN/docs/Web/CSS/letter-spacing
   */
  spacing: e(),
  /**
   * 行高
   *
   * @see line-height https://developer.mozilla.org/zh-CN/docs/Web/CSS/line-height
   */
  lineHeight: e(),
  /**
   * 首字符缩进量
   *
   * @see text-indent https://developer.mozilla.org/zh-CN/docs/Web/CSS/text-indent
   */
  indent: e(),
  /**
   * 文字是否以粗体显示
   *
   * @see font-weight https://developer.mozilla.org/zh-CN/docs/Web/CSS/font-weight
   */
  bold: r(),
  /**
   * 文字的修饰
   *
   * @see text-decoration https://developer.mozilla.org/zh-CN/docs/Web/CSS/text-decoration
   * @values overline line-through underline
   */
  decoration: o(void 0, (i) => ["overline", "line-through", "underline"].includes(i)),
  /**
   * 内边距
   *
   * @see padding https://developer.mozilla.org/zh-CN/docs/Web/CSS/padding
   */
  padding: e(),
  /**
   * 宽度
   *
   * @see width https://developer.mozilla.org/zh-CN/docs/Web/CSS/width
   */
  width: e(),
  /** 是否省略文本，需配合 width 宽度使用 */
  ellipsis: r(),
  /**
   * 是否居中
   *
   * @see text-align https://developer.mozilla.org/zh-CN/docs/Web/CSS/text-align
   */
  center: r()
};
export {
  l as Props
};
