// fighting-design/es/_hooks/use-count-down/index.js
import { ref as k, computed as F, onBeforeUnmount as U, onActivated as g, onDeactivated as x } from "vue";
import { isBrowser as u } from "../../_utils/is/index.js";
const T = (e) => u ? window.requestAnimationFrame(e) : -1, B = (e) => {
  u && window.cancelAnimationFrame(e);
}, i = 1e3, f = 60 * i, l = 60 * f, R = 24 * l, E = (e) => {
  const r = Math.floor(e / R), a = Math.floor(e % R / l), t = Math.floor(e % l / f), c = Math.floor(e % f / i), n = Math.floor(e % i);
  return {
    total: e,
    days: r,
    hours: a,
    minutes: t,
    seconds: c,
    milliseconds: n
  };
}, I = (e, r, a = i) => Math.floor(e / a) === Math.floor(r / a), S = (e) => {
  let r, a, t, c;
  const n = k(e.time), A = F(() => E(n.value)), m = () => Math.max(a - performance.now(), 0), d = (o) => {
    var w;
    n.value = o, o === 0 && (s(), (w = e.onFinish) == null || w.call(e));
  }, h = () => {
    u && (e.millisecond ? v() : M());
  }, C = () => {
    t || (a = performance.now() + n.value, t = !0, h());
  }, D = (o = e.time) => {
    s(), n.value = o;
  }, s = () => {
    t = !1, B(r);
  }, v = () => {
    r = T(() => {
      t && (d(m()), n.value > 0 && v());
    });
  }, M = () => {
    r = T(() => {
      if (t) {
        const o = m();
        (!I(o, n.value, e.interval) || o === 0) && d(o), n.value > 0 && M();
      }
    });
  };
  return U(s), g(() => {
    c && (t = !0, c = !1, h());
  }), x(() => {
    t && (s(), c = !0);
  }), {
    start: C,
    pause: s,
    reset: D,
    current: A
  };
};
export {
  S as useCountDown
};
