// fighting-design/es/close-btn/src/index.js
import { defineComponent as p, openBlock as f, createElementBlock as u, normalizeClass as d, unref as e, normalizeStyle as C, createVNode as t, withCtx as b, renderSlot as k } from "vue";
import { Props as z } from "./index3.js";
import { FSvgIcon as h } from "../../svg-icon/index.js";
import v from "../../_svg/f-icon-cross/index.js";
import { useList as y } from "../../_hooks/use-list/index.js";
import { useRun as _ } from "../../_hooks/use-run/index.js";
var E = /* @__PURE__ */ p({
  name: "FCloseBtn",
  __name: "close-btn",
  props: z,
  setup(n) {
    const s = n, { styles: r, classes: l } = y(s, "close-btn"), { run: c } = _(), i = (o) => {
      s.disabled || c(s.onClick, o);
    }, a = l(["round", "disabled"], "f-close-btn"), m = r(["color", "hoverColor"]);
    return (o, B) => (f(), u("div", {
      role: "button",
      class: d(e(a)),
      style: C(e(m)),
      onClick: i
    }, [
      t(e(h), {
        size: o.size,
        icon: o.icon
      }, {
        default: b(() => [
          k(o.$slots, "default", {}, () => [
            t(e(v))
          ])
        ]),
        _: 3
      }, 8, ["size", "icon"])
    ], 6));
  }
});
export {
  E as default
};
