// fighting-design/es/pagination/src/index.js
import { defineComponent as $, mergeModels as G, useModel as S, ref as j, openBlock as l, createElementBlock as a, normalizeClass as s, unref as e, createBlock as f, withCtx as B, Fragment as y, renderList as V, createCommentVNode as u, createElementVNode as b, createVNode as d, toDisplayString as z, isRef as R } from "vue";
import { Props as U } from "./index3.js";
import { FInput as D } from "../../input/index.js";
import { FSelect as X } from "../../select/index.js";
import { FOption as q } from "../../option/index.js";
import { FSvgIcon as m } from "../../svg-icon/index.js";
import { usePage as H } from "../../_hooks/use-page/index.js";
import { useTurnPage as J } from "../../_hooks/use-turn-page/index.js";
import { PAGINATION as K, PAGINATION_ITEM as r, PAGINATION_PREV as Q, PAGINATION_NEXT as W } from "../../_tokens/key/index.js";
import Y from "../../_svg/f-icon-chevron-left/index.js";
import M from "../../_svg/f-icon-menu-meatball/index.js";
import Z from "../../_svg/f-icon-chevron-right/index.js";
import { useList as x } from "../../_hooks/use-list/index.js";
const _ = ["disabled"], ee = ["onClick"], ne = ["disabled"];
var fe = /* @__PURE__ */ $({
  name: "FPagination",
  __name: "pagination",
  props: /* @__PURE__ */ G(U, {
    current: {
      default: 1,
      type: Number
    },
    currentModifiers: {},
    pageSize: {
      default: 10,
      type: Number
    },
    pageSizeModifiers: {}
  }),
  emits: ["update:current", "update:pageSize"],
  setup(i) {
    const p = i, v = S(i, "current"), k = S(i, "pageSize"), { pages: I, showNextMore: P, showPrevMore: F, maxCount: c, handelTurnPages: C } = H(p, {
      currentModelValue: v,
      pageSizeModelValue: k
    }), { jumpCurrent: g, selectChange: A, handelChange: T, handleInput: N, handelClick: w } = J(p, { currentModelValue: v, pageSizeModelValue: k }, I, c), { classes: E } = x(p, "pagination"), h = j(10), L = E(["background", "round", "disabled"], K);
    return (o, t) => (l(), a("div", {
      class: s(e(L))
    }, [
      o.pageSelect && o.pageSelect.length ? (l(), f(e(X), {
        key: 0,
        modelValue: h.value,
        "onUpdate:modelValue": t[0] || (t[0] = (n) => h.value = n),
        width: 120,
        disabled: o.disabled,
        "on-change": e(A)
      }, {
        default: B(() => [
          (l(!0), a(y, null, V(o.pageSelect, (n) => (l(), f(e(q), {
            key: n,
            value: n,
            label: n + "/页"
          }, null, 8, ["value", "label"]))), 128))
        ]),
        _: 1
      }, 8, ["modelValue", "disabled", "on-change"])) : u("", !0),
      b("button", {
        class: "f-pagination__button",
        disabled: o.disabled,
        onClick: t[1] || (t[1] = (n) => e(C)("prev"))
      }, [
        d(e(m), {
          size: 14,
          icon: o.prevIcon || e(Y)
        }, null, 8, ["icon"])
      ], 8, _),
      o.total > 0 ? (l(), a("div", {
        key: 1,
        class: "f-pagination__pages",
        onClick: t[2] || (t[2] = (n) => e(w)(n))
      }, [
        b("div", {
          class: s([
            e(r),
            {
              [`${e(r)}-active`]: i.current === 1
            }
          ])
        }, " 1 ", 2),
        e(F) ? (l(), a("div", {
          key: 0,
          class: s([e(r), e(Q)])
        }, [
          d(e(m), {
            size: 15,
            icon: e(M)
          }, null, 8, ["icon"])
        ], 2)) : u("", !0),
        (l(!0), a(y, null, V(e(I), (n) => (l(), a("div", {
          key: n,
          class: s([
            e(r),
            {
              [`${e(r)}-active`]: i.current === n
            }
          ]),
          onClick: (O) => e(T)(n, O)
        }, z(n), 11, ee))), 128)),
        e(P) ? (l(), a("div", {
          key: 1,
          class: s([e(r), e(W)])
        }, [
          d(e(m), {
            size: 15,
            icon: e(M)
          }, null, 8, ["icon"])
        ], 2)) : u("", !0),
        o.total > 1 && o.total > Number(i.pageSize) ? (l(), a("div", {
          key: 2,
          class: s([
            e(r),
            {
              [`${e(r)}-active`]: i.current === e(c)
            }
          ])
        }, z(e(c)), 3)) : u("", !0)
      ])) : u("", !0),
      b("button", {
        class: "f-pagination__button",
        disabled: o.disabled,
        onClick: t[3] || (t[3] = (n) => e(C)("next"))
      }, [
        d(e(m), {
          size: 14,
          icon: o.nextIcon || e(Z)
        }, null, 8, ["icon"])
      ], 8, ne),
      o.jumpSearch ? (l(), f(e(D), {
        key: 2,
        modelValue: e(g),
        "onUpdate:modelValue": t[4] || (t[4] = (n) => R(g) ? g.value = n : null),
        placeholder: "输入跳转的页数",
        class: "f-pagination__jump",
        style: { width: "80px" },
        disabled: o.disabled,
        "on-blur": e(N),
        "on-enter": e(N)
      }, null, 8, ["modelValue", "disabled", "on-blur", "on-enter"])) : u("", !0)
    ], 2));
  }
});
export {
  fe as default
};
