// fighting-design/es/input/src/index3.js
import { setStringNumberProp as t, setStringProp as e, setBooleanProp as o, setNumberProp as s, setObjectProp as a, setFunctionProp as r } from "../../_utils/props/index.js";
import { FIGHTING_SIZE as p } from "../../_tokens/attrs/index.js";
const l = {
  /** 绑定值 */
  modelValue: t(),
  /**
   * 类型
   *
   * 文字输入框或者密码输入框
   *
   * @values text password number
   * @default text
   */
  type: e("text", (n) => ["text", "password", "number"].includes(n)),
  /**
   * 尺寸
   *
   * @values large middle small mini
   */
  size: e(void 0, (n) => p.includes(n)),
  /**
   * 是否禁用
   *
   * @see disabled https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/Input#attr-disabled
   */
  disabled: o(),
  /**
   * 最大（数字或日期时间）值
   *
   * @see max https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/Input#attr-max
   */
  max: s(),
  /**
   * 最小（数字或日期时间）值
   *
   * @see min https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/Input#attr-min
   */
  min: s(),
  /**
   * 最大输入长度
   *
   * @see maxLength https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/Input#attr-maxlength
   */
  maxLength: s(),
  /**
   * 设置字体大小
   *
   * @see font-size https://developer.mozilla.org/zh-CN/docs/Web/CSS/font-size
   */
  fontSize: t(),
  /**
   * 是否自动获取焦点
   *
   * @see autofocus https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/Input#attr-autofocus
   */
  autofocus: o(),
  /**
   * 原生 name 属性
   *
   * @see name https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/Input#attr-name
   */
  name: e("f-input"),
  /**
   * 占位符
   *
   * @see placeholder https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/Input#attr-placeholder
   */
  placeholder: e(),
  /** 占位符颜色 */
  placeholderColor: e(),
  /** 是否可清除 */
  clear: o(),
  /** 自定义宽度 */
  width: t(),
  /** 自定义高度 */
  height: t(),
  /** 是否可搜索 */
  search: o(),
  /** 自定义文字颜色 */
  textColor: e(),
  /**
   * 是否只读
   *
   * @see readonly https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/Input#attr-readonly
   */
  readonly: o(),
  /** 是否显示查看密码的按钮 */
  showPassword: o(),
  /** 是否按下回车进行搜索 */
  enterSearch: o(),
  /** 自定义 icon */
  icon: a(),
  /** 后缀 icon */
  afterIcon: a(),
  /**
   * 是否开启自动填充特性提示
   *
   * 注意这不是一个布尔属性！
   *
   * 详情配置可参考
   *
   * @see HTML属性autocomplete https://developer.mozilla.org/zh-CN/docs/Web/HTML/Attributes/autocomplete
   * @see autocomplete https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/Input#autocomplete
   */
  autocomplete: e("off"),
  /** 点击搜索之后触发的回调 */
  onSearch: r(),
  /** 绑定值发生改变时触发的回调 */
  onChange: r(),
  /** input 事件触发的回调 */
  onInput: r(),
  /** 失去焦点时触发的回调 */
  onBlur: r(),
  /** 获得焦点时触发的回调 */
  onFocus: r(),
  /** 按下回车触发的 */
  onEnter: r()
};
export {
  l as Props
};
