// fighting-design/es/confirm-box/src/index.js
import { defineComponent as k, ref as d, openBlock as m, createBlock as y, Teleport as g, createVNode as a, Transition as B, withCtx as r, createElementBlock as w, createElementVNode as n, toDisplayString as s, unref as l, createTextVNode as _, createCommentVNode as E } from "vue";
import { Props as F } from "./index3.js";
import { FButton as u } from "../../button/index.js";
import { FSpace as T } from "../../space/index.js";
import { FCloseBtn as N } from "../../close-btn/index.js";
import { isFunction as p } from "../../_utils/is/index.js";
import { useRun as V } from "../../_hooks/use-run/index.js";
const S = {
  key: 0,
  class: "f-confirm-box"
}, A = /* @__PURE__ */ n("div", { class: "f-confirm-box__mask" }, null, -1), L = { class: "f-confirm-box__container" }, O = { class: "f-confirm-box__header" }, D = { class: "f-confirm-box__title" }, P = { class: "f-confirm-box__body" }, R = { class: "f-confirm-box__footer" };
var M = /* @__PURE__ */ k({
  name: "FConfirmBox",
  __name: "confirm-box",
  props: F,
  setup(v) {
    const e = v, { run: f } = V(), i = d(e.show), t = d(!1), c = () => {
      i.value = !1;
    }, h = () => {
      f(e.onOpen, i.value);
    }, C = () => {
      f(e.onClose, i.value);
    }, b = async (o) => {
      t.value = !0, p(e.onConfirm) && await e.onConfirm(o), c();
    }, x = async (o) => {
      t.value = !0, p(e.onCancel) && await e.onCancel(o), c();
    };
    return (o, j) => (m(), y(g, { to: "body" }, [
      a(B, {
        name: "f-confirm-box__trans",
        appear: "",
        onAfterEnter: h,
        onAfterLeave: C
      }, {
        default: r(() => [
          i.value ? (m(), w("div", S, [
            A,
            n("div", L, [
              n("div", O, [
                n("div", D, s(o.title), 1),
                a(l(N), {
                  disabled: t.value,
                  "on-click": c
                }, null, 8, ["disabled"])
              ]),
              n("div", P, s(o.content), 1),
              n("div", R, [
                a(l(T), null, {
                  default: r(() => [
                    a(l(u), {
                      loading: t.value,
                      "on-click": x
                    }, {
                      default: r(() => [
                        _(s(o.cancelText || "取消"), 1)
                      ]),
                      _: 1
                    }, 8, ["loading"]),
                    a(l(u), {
                      loading: t.value,
                      type: "primary",
                      "on-click": b
                    }, {
                      default: r(() => [
                        _(s(o.confirmText || "确定"), 1)
                      ]),
                      _: 1
                    }, 8, ["loading"])
                  ]),
                  _: 1
                })
              ])
            ])
          ])) : E("", !0)
        ]),
        _: 1
      })
    ]));
  }
});
export {
  M as default
};
