// fighting-design/es/image/src/index3.js
import { setStringProp as r, setBooleanProp as o, setStringNumberProp as t, setFunctionProp as e } from "../../_utils/props/index.js";
import { FIGHTING_FIT as n } from "../../_tokens/attrs/index.js";
const s = {
  /** 图片路径 */
  src: r(),
  /**
   * 原生 alt 属性
   *
   * @see alt https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/img#attr-alt
   */
  alt: r(),
  /** 是否可以拖动 */
  draggable: o(!0),
  /** 是否懒加载图片 */
  lazy: o(),
  /** 触发懒加载的距离 */
  rootMargin: t("100px"),
  /** 图片的宽度 */
  width: t(),
  /** 图片的高度 */
  height: t(),
  /** 是否为块级元素 */
  block: o(),
  /**
   * 如何适应容器，原生 object-fit 属性
   *
   * @values fill contain cover none scale-down
   * @see object-fit https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit#try_it
   */
  fit: r(void 0, (i) => n.includes(i)),
  /** 是否可以选中 */
  select: o(),
  /**
   * 原生 referrerPolicy 属性
   *
   * @see Referrer-Policy https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referrer-Policy#syntax
   */
  referrerPolicy: r(),
  /** 配置圆角，传入一个数值 */
  round: t(),
  /** 加载失败时要显示的地址 */
  errSrc: r(),
  /** img 的 title */
  title: r(),
  /** 图片加载成功触发的回调 */
  onLoad: e(),
  /** 图片加载失败触发的回调 */
  onError: e()
};
export {
  s as Props
};
