// fighting-design/es/select/src/index3.js
import { setStringProp as e, setBooleanProp as r, setFunctionProp as o } from "../../_utils/props/index.js";
import { FIGHTING_SIZE as l } from "../../_tokens/attrs/index.js";
const p = {
  /** 绑定值 */
  modelValue: {
    type: [String, Number, Boolean],
    default: () => null,
    required: !0
  },
  /** 自定义宽度 */
  width: {
    type: [Number, String],
    default: () => 190
  },
  /**
   * 原生 name 属性
   *
   * @see name https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/Input#attr-name
   */
  name: e("f-select"),
  /**
   * 尺寸
   *
   * @values large middle small mini
   * @default middle
   */
  size: e("middle", (t) => l.includes(t)),
  /**
   * 占位符
   *
   * @see placeholder https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/Input#attr-placeholder
   */
  placeholder: e(),
  /** 是否可清除 */
  clear: r(),
  /** 是否可模糊搜索过滤 */
  filter: r(),
  /**
   * 是否禁用
   *
   * @see disabled https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/Input#attr-disabled
   */
  disabled: r(),
  /** 绑定值发生变化时触发的回调 */
  onChange: o()
}, s = Symbol("f-select-props-key");
export {
  p as Props,
  s as SELECT_PROPS_TOKEN
};
