// fighting-design/es/card/src/index.js
import { defineComponent as y, ref as k, openBlock as o, createBlock as l, Transition as v, withCtx as B, createElementBlock as n, normalizeClass as d, unref as r, normalizeStyle as b, createElementVNode as c, renderSlot as i, createTextVNode as g, toDisplayString as S, createCommentVNode as s } from "vue";
import { Props as w } from "./index3.js";
import { FCloseBtn as N } from "../../close-btn/index.js";
import { useList as V } from "../../_hooks/use-list/index.js";
import { useRun as $ } from "../../_hooks/use-run/index.js";
const z = {
  key: 0,
  role: "heading",
  class: "f-card__header"
}, E = {
  role: "none",
  class: "f-card__body"
};
var j = /* @__PURE__ */ y({
  name: "FCard",
  __name: "card",
  props: w,
  setup(m) {
    const a = m, t = k(!0), { run: u } = $(), { classes: f, styles: p } = V(a, "card"), h = f(["round", "shadow"], "f-card"), _ = p(["padding", "borderColor", "background", "titleColor"]), C = () => {
      t.value = !1, u(a.onClose, t.value);
    };
    return (e, F) => (o(), l(v, { name: "f-card" }, {
      default: B(() => [
        t.value ? (o(), n("div", {
          key: 0,
          class: d(r(h)),
          style: b(r(_))
        }, [
          e.$slots.header || e.title ? (o(), n("header", z, [
            c("span", {
              class: d(["f-card__title", { "f-card__title-bold": e.titleBold }])
            }, [
              i(e.$slots, "header", {}, () => [
                g(S(e.title), 1)
              ])
            ], 2),
            e.close ? (o(), l(r(N), {
              key: 0,
              "on-click": C
            })) : s("", !0)
          ])) : s("", !0),
          c("main", E, [
            i(e.$slots, "default")
          ])
        ], 6)) : s("", !0)
      ]),
      _: 3
    }));
  }
});
export {
  j as default
};
