// fighting-design/es/menu/src/index.js
import { defineComponent as f, ref as d, computed as _, provide as p, toRef as v, openBlock as t, createElementBlock as s, normalizeClass as k, unref as C, renderSlot as n, createCommentVNode as m, createElementVNode as b } from "vue";
import { Props as E, MENU_PROVIDE_KEY as N } from "./index3.js";
import { useList as $ } from "../../_hooks/use-list/index.js";
const h = {
  key: 0,
  class: "f-menu__before"
}, M = { class: "f-menu__content" }, I = {
  key: 1,
  class: "f-menu__after"
};
var B = /* @__PURE__ */ f({
  name: "FMenu",
  __name: "menu",
  props: E,
  setup(a) {
    const o = a, { classes: c } = $(o, "menu"), r = d(o.activeName), l = _(() => r.value), u = (e) => {
      r.value = e;
    };
    p(N, {
      mode: v(o, "mode"),
      defaultActive: l,
      setActiveName: u,
      onMenuItemClick: o.onMenuItemClick,
      onSubmenuClick: o.onSubmenuClick
    });
    const i = c(["mode"], "f-menu");
    return (e, S) => (t(), s("div", {
      role: "menu",
      class: k(C(i))
    }, [
      e.$slots.before ? (t(), s("div", h, [
        n(e.$slots, "before")
      ])) : m("", !0),
      b("div", M, [
        n(e.$slots, "default")
      ]),
      e.$slots.after ? (t(), s("div", I, [
        n(e.$slots, "after")
      ])) : m("", !0)
    ], 2));
  }
});
export {
  B as default
};
