// fighting-design/es/input-number/src/index3.js
import { setNumberProp as e, setStringProp as n, setBooleanProp as t, setFunctionProp as r } from "../../_utils/props/index.js";
import { FIGHTING_TYPE as u, FIGHTING_SIZE as i } from "../../_tokens/attrs/index.js";
const d = {
  /**
   * 绑定值
   *
   * @default 1
   */
  modelValue: e(0),
  /**
   * 模式
   *
   * @values default button switch
   * @default default
   */
  model: n("default", (o) => ["default", "button", "switch"].includes(o)),
  /**
   * 数值精度
   *
   * @default 0
   */
  precision: e(0),
  /** 切换的按钮类型 */
  /**
   * 按钮的类型（非自定义按钮颜色时有效）
   *
   * @values default primary success danger warning
   */
  buttonType: n(void 0, (o) => u.includes(o)),
  /**
   * 步长
   *
   * @default 1
   */
  step: e(1),
  /**
   * 尺寸
   *
   * @values large middle small mini
   * @default middle
   */
  size: n("middle", (o) => i.includes(o)),
  /** 是否禁用 */
  disabled: t(),
  /** 最大值 */
  max: e(),
  /** 最小值 */
  min: e(),
  /** 最大输入长度 */
  maxLength: e(),
  /** 是否自动聚焦 */
  autofocus: t(),
  /**
   * 原生 name 属性
   *
   * @see name https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/Input#attr-name
   */
  name: n("f-input-number"),
  /**
   * 占位符
   *
   * @see placeholder https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/Input#attr-placeholder
   */
  placeholder: n(),
  /** 是否可清除 */
  clear: t(),
  /** 是否只读 */
  readonly: t(),
  /** 绑定值发生改变时触发的回调 */
  onChange: r(),
  /** 失去焦点时触发的回调 */
  onBlur: r(),
  /** 获得焦点时触发的回调 */
  onFocus: r(),
  /** input 事件触发的回调 */
  onInput: r()
};
export {
  d as Props
};
